cpp/src/arrow/compute/exec/aggregate_node.cc

```diff
@@ -697,16 +925,22 @@ class GroupByNode : public ExecNode, public TracedNode {
   }
 
   int output_task_group_id_;
+  std::unique_ptr<GroupingSegmenter> segmenter_;
+  std::vector<Datum> segmenter_values_;
 
   const std::vector<int> key_field_ids_;
+  const std::vector<int> segment_key_field_ids_;
+  const std::vector<std::vector<TypeHolder>> agg_src_types_;
   const std::vector<std::vector<int>> agg_src_fieldsets_;
   const std::vector<Aggregate> aggs_;
-  const std::vector<const HashAggregateKernel*> agg_kernels_;
+  std::vector<const HashAggregateKernel*> agg_kernels_;
```

Comment:
Why removing const?

Reply:
Because `agg_kernels_` is reset in `ReconstructAggregates()`.

Reply:
why do we need to reset kernels? Is resetting kernel states not enough?

Reply:
This is now fixed.